contents/staticAnalysis/ESLint.md

```diff
@@ -0,0 +1,104 @@
+# ESLint
+
+Author: [Nicholas Chua](https://github.com/nus-cs3281/2018/blob/master/students/nicholasChua/nicholasChua-Resume.md)
+
+## Overview
+
+[ESLint](https://eslint.org/docs/about/) is an open source JavaScript linter.
+
+Code linting is a type of static analysis that helps find problematic patterns or code that does not follow a certain style guideline.
+
+JavaScript, being a loosely-typed, interpreted language, is prone to developer errors and therefore linting is especially important when working with JavaScript.
+
+## Features
+
+### In Built Rules
+[In built rules](https://eslint.org/docs/rules/) in ESLint are split into a few categories:
+
+* Possible Errors
+* Best Practices
+* Variables
+* Strict Mode
+* Node.js and CommonJS
+* Stylistic Issues
+* ECMAScript 6 (ES6)
+
+### Custom Rules
+Custom rules can be imported via Plugins. You can [learn how to write your own](https://eslint.org/docs/developer-guide/working-with-plugins) or [use existing ones](https://www.npmjs.com/browse/keyword/eslintplugin) that are already published to npm.
+
+### Integrations
+ESLint has [Editor Integrations](https://eslint.org/docs/user-guide/integrations#editors) such like syntax highlighting in many popular editors.
+
+## How to use it
+
+### Installation
+ESLint requires Node.js (>= 4.x) and npm version 2+. You can [install it with npm](https://eslint.org/docs/user-guide/getting-started#installation-and-usage).
+
+### Configuration
+#### Initial Configuration
+ESLint can be configured in one of two ways.
+
+1. Configuration Comments
+1. Configuration File
+
+Both of which are explained [here](https://eslint.org/docs/user-guide/configuring#configuring-rules). You can learn how to set it up [here](https://eslint.org/docs/user-guide/configuring).
+
+NOTE: If you ran `$ /node_modules/.bin/eslint --init` (as advised by ESLint installation guide), you will be guided by a CLI to create an `.eslintrc` configuration file.
+
+
+#### Rules
+ESLint comes shipped with a large number of rules. You can [configure the rules](https://eslint.org/docs/user-guide/configuring#configuring-rules) to suit your needs.
+
+If you are unsure about the configuration options of the rule you want to change, you can refer to the comprehensive [rule documentation](https://eslint.org/docs/rules/).
+
+#### Plugins
+As mentioned earlier, you can use third-party plugins to have your own custom rules. To import them, [include them inside your configuration file](https://eslint.org/docs/user-guide/configuring).
+
+A popular plugin would be [eslint-plugin-lodash](https://github.com/wix/eslint-plugin-lodash) which is meant to enforce good practices and styling with the utility library [Lodash](https://lodash.com/).
+

```

Comment:
Maybe you can share some good plugins e.g. for react?
Possible list [here](https://github.com/dustinspecker/awesome-eslint)

Reply:
Thanks for the link, i've placed it under resources. I think placing a plugin for React may not be suitable for everyone.